### PR TITLE
Fixed the lua syntax highlighter once and for all. (Redone)

### PR DESCRIPTION
**Pull request by ccuser44**
This is a redone version of #12367

The Npp Lua syntax highlighter is known for being buggy and hard to develop with.

For some reason the Lua language xml file has the Corona SDK globals pasted in to it. This makes it so that the syntax highlighter makes incorrect suggestions suggesting totally irrelevant funtions from a 3rd party library which makes it cumbersome to use the typing suggestions and is confusing for new users.

Not to mention the Corona SDK including dublicate functions which bricks the suggestor preventing it from showing the arguments & description of the default global functions.

Here are the changes made here:

- Removed Corona SDK & Dublicate default globals
- Seperated different keywords to make it easier to update this in the future
- Added all of the keywords from the newer Lua versions

I'm not sure if there are bugs however. I manually copied the newer keywords from https://www.lua.org/manual/ so there might be a few missing keywords or something else might be incorrect.

### CODE REVIEW

**Comment by ccuser44** [comment]
Ok I have re-done #12367 without the accidental formatting changes

**Comment by ccuser44** [comment]
I'm not sure if I missed some newer Lua globals though because I added all of these manually.

**Comment by donho** [comment]
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/12367#issuecomment-1314492403

File: PowerEditor/installer/APIs/lua.xml [hunk 1]
```diff
@@ -1220,24 +1237,24 @@ captures from the pattern; otherwise it returns nil. If pattern specifies no cap
 the whole match is returned. A third, optional numerical argument init specifies where to 
 start the search; its default value is 1 and can be negative. 
         ">
-                <Param name="s" />
-                <Param name="pattern" />
-                <Param name="[, init]" />
+                <Param name="s"/>
+                <Param name="pattern"/>
+                <Param name="[, init]"/>
```
Comment (donho) [inline PowerEditor/installer/APIs/lua.xml:1242]: It's not necessary to remove all the white space.
Reformatting is not allowed, as I mentioned:
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/12367#issuecomment-1314492403

Comment (ccuser44) [reply PowerEditor/installer/APIs/lua.xml:1242]: So do I need to fix the small whitespace additions as well?

**Pull request closed without merging**